### PR TITLE
Adding ability to install previews side-by-side with production releases

### PR DESCRIPTION
**Pull request by DarwinJS**
## PR Summary
-preview switch:
 - installs pwsh-preview packages when a package manager is being used.
 - uses link name pwsh-preview when tar ball install is being done
 - results in preview release being available for calling manually, but does not become systemwide version
- Issue #7177

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed - Issue link: https://github.com/PowerShell/PowerShell/issues/7177
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/install-powershell-readme.md [hunk 1]
```diff
@@ -24,7 +24,7 @@
 * -includeide - installs vscode and vscode powershell extension (only relevant to machines with desktop environment)
 * -interactivetesting - do a quick launch test of vscode - only relevant when used with -includeide
 * -skip-sudo-check - use sudo without verifying it's availability (hard to accurately do on some distros)
-* -allowprerelease - includes prereleases of powershell core when selection allows this.  Repository based installs take the latest on the repo url regardless of this switch.
```
Comment (TravisEz13) [inline tools/install-powershell-readme.md]: technically, this is a breaking change, but not in the product

Comment (TravisEz13) [reply tools/install-powershell-readme.md]: I think it is fine since it hasn't been in for long

Comment (DarwinJS) [reply tools/install-powershell-readme.md]: Yeah, I think it was only working for 24-36 hours.

Comment (TravisEz13) [reply tools/install-powershell-readme.md]: Agreed, this lgtm, but I'm supposed to give 24 hours for review.

Comment (DarwinJS) [reply tools/install-powershell-readme.md]: Also these changes are just trying to keep up with significant "breaking change" design decisions - like establishing the entirely separate package id for previews.  It's a great change - but such shifts are the lions share of "breaking" compared to the breaking code changes needed to support them.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**